auth-layer-proxy/scripts/start-envoy.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+
+# Set default values for environment variables if they are not already set
+export SERVICE_TYPE=${SERVICE_TYPE:-LOGICAL_DNS}
+export SERVICE_ADDRESS=${SERVICE_ADDRESS:-host.docker.internal}
+export SERVICE_PORT=${SERVICE_PORT:-8020}
+export ENVOY_ADMIN_PORT=${ENVOY_ADMIN_PORT:-15000}
+export PROXY_PORT=${PROXY_PORT:-10000}
+export CLIENT_ID=${CLIENT_ID:-htg-auth-layer}
+export TOKEN_INTROSPECTION_URL=${TOKEN_INTROSPECTION_URL:-http://host.docker.internal:8080/realms/HederaTheGraph/protocol/openid-connect/token/introspect}
+
+# print the environment variables
+echo "----------------------------------------"
+echo "---  Environment variables:          ---"
+echo "----------------------------------------"
+echo "SERVICE_TYPE: $SERVICE_TYPE"
+echo "SERVICE_ADDRESS: $SERVICE_ADDRESS"
+echo "SERVICE_PORT: $SERVICE_PORT"
+echo "ENVOY_ADMIN_PORT: $ENVOY_ADMIN_PORT"
+echo "PROXY_PORT: $PROXY_PORT"
+echo "CLIENT_ID: $CLIENT_ID"
+# Extract the first 4 characters of CLIENT_SECRET
+CLIENT_SECRET_FIRST_4=$(echo $CLIENT_SECRET | cut -c 1-4)
+echo "CLIENT_SECRET: ${CLIENT_SECRET_FIRST_4}******"
+echo "TOKEN_INTROSPECTION_URL: $TOKEN_INTROSPECTION_URL"
+echo "----------------------------------------"
+
+
+# Use envsubst to replace environment variables in the template
+envsubst < /etc/envoy/configs/envoy-auth-template.yaml > /etc/envoy/envoy-config.yaml
+
+# Print the processed configuration
+echo "----------------------------------------"
+echo "---  Processed Envoy configuration:  ---"
+echo "----------------------------------------"
+cat /etc/envoy/envoy-config.yaml
```

Comment:
Q: Just checking that this doesn't defeat the purpose of the first 4 chars i.e. does this cat expose the secret in logs also?

Reply:
no it does not, the client_secret is used directy on the `TokenVerificationFilter.lua`
https://github.com/hashgraph/hedera-the-graph/blob/main/auth-layer-proxy/filters/TokenVerificationFilter.lua#L25

the output of this is useful for troubleshooting and verifying that the configuration was processed as desired and it looks like this:

```yaml
envoy-1  | ----------------------------------------
envoy-1  | ---  Environment variables:          ---
envoy-1  | ----------------------------------------
envoy-1  | SERVICE_TYPE: LOGICAL_DNS
envoy-1  | SERVICE_ADDRESS: host.docker.internal
envoy-1  | SERVICE_PORT: 8020
envoy-1  | ENVOY_ADMIN_PORT: 16000
envoy-1  | PROXY_PORT: 10000
envoy-1  | CLIENT_ID: htg-auth-layer
envoy-1  | CLIENT_SECRET: 0cyY******
envoy-1  | TOKEN_INTROSPECTION_URL: http://host.docker.internal:8080/realms/HederaTheGraph/protocol/openid-connect/token/introspect
envoy-1  | ----------------------------------------
envoy-1  | ----------------------------------------
envoy-1  | ---  Processed Envoy configuration:  ---
envoy-1  | ----------------------------------------
envoy-1  | admin:
envoy-1  |   address:
envoy-1  |     socket_address:
envoy-1  |       address: 0.0.0.0
envoy-1  |       port_value: 16000
envoy-1  | 
envoy-1  | static_resources:
envoy-1  |   clusters:
envoy-1  |   - name: upstream_service
envoy-1  |     connect_timeout: 5s
envoy-1  |     type: LOGICAL_DNS
envoy-1  |     load_assignment:
envoy-1  |       cluster_name: upstream_service
envoy-1  |       endpoints:
envoy-1  |       - lb_endpoints:
envoy-1  |         - endpoint:
envoy-1  |             address:
envoy-1  |               socket_address:
envoy-1  |                 address: host.docker.internal
envoy-1  |                 port_value: 8020
envoy-1  | 
envoy-1  |   listeners:
envoy-1  |   - name: listener_0
envoy-1  |     address:
envoy-1  |       socket_address:
envoy-1  |         address: 0.0.0.0
envoy-1  |         port_value: 10000
envoy-1  |     filter_chains:
envoy-1  |     - filters:
envoy-1  |       - name: envoy.filters.network.http_connection_manager
envoy-1  |         typed_config:
envoy-1  |           "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
envoy-1  |           stat_prefix: edge
envoy-1  |           http_filters:
envoy-1  |           - name: envoy.filters.http.lua
envoy-1  |             typed_config:
envoy-1  |               "@type": type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua
envoy-1  |               default_source_code: 
envoy-1  |                 filename: /etc/envoy/filters/TokenVerificationFilter.lua
envoy-1  | 
envoy-1  |           - name: envoy.filters.http.router
envoy-1  |             typed_config:
envoy-1  |               "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router          
envoy-1  |           route_config:
envoy-1  |             virtual_hosts:
envoy-1  |             - name: all_domains
envoy-1  |               domains: ["*"]
envoy-1  |               routes:
envoy-1  |               - match:
envoy-1  |                   prefix: "/"
envoy-1  |                   headers:
envoy-1  |                     name: ":method"
envoy-1  |                     exact_match: "POST"
envoy-1  |                 route:
envoy-1  |                   cluster: upstream_service
envoy-1  | ----------------------------------------
```